resolver/query_logging_resolver.go

```diff
@@ -125,6 +143,42 @@ func (r *QueryLoggingResolver) Resolve(request *model.Request) (*model.Response,
 	return resp, err
 }
 
+func (r *QueryLoggingResolver) createLogEntry(request *model.Request, response *model.Response,
+	start time.Time, durationMs int64) *querylog.LogEntry {
+	entry := querylog.LogEntry{
+		Start:       start,
+		ClientIP:    "0.0.0.0",
+		ClientNames: []string{"none"},
+	}
+
+	for _, f := range r.fields {
+		switch f {
+		case config.QueryLogFieldClientIP:
+			entry.ClientIP = request.ClientIP.String()
+
+		case config.QueryLogFieldClientName:
+			entry.ClientNames = request.ClientNames
+
+		case config.QueryLogFieldResponseReason:
+			entry.ResponseReason = response.Reason
+			entry.ResponseType = response.RType.String()
+			entry.ResponseCode = dns.RcodeToString[response.Res.Rcode]
+
+		case config.QueryLogFieldResponseAnswer:
+			entry.Answer = util.AnswerToString(response.Res.Answer)
+
+		case config.QueryLogFieldQuestion:
+			entry.QuestionName = request.Req.Question[0].Name
```

Comment:
should use util.QuestionToString -> this will obfuscate question if log privacy is enabled